### ISSUE TITLE
WithSvg panics when arc is centered at the origin

### ISSUE THREAD

**Issue by Nilirad** [nical/lyon#650]
Trying to draw an arc centered at the origin causes a panic. Choosing other points don't.

Example:

```
use lyon_tessellation::{
    math::{point, vector, Angle},
    path::{builder::WithSvg, path::Builder},
};

fn main() {
    use std::f32::consts::PI;
    let mut b = WithSvg::new(Builder::new());
    b.arc(
        point(0.0, 0.0),
        vector(50.0, 50.0),
        Angle::radians(PI),
        Angle::radians(0.0),
    );
}
```

**Comment by nical** [comment]
Hi, thanks for reporting this and providing a nice and simple test case. This is caused by the center and starting point of the arc being equal (resulting in NaN when computing the angle of a nil vector). It looks like in a similar situation, inkscape just creates a straight line so handle this case that way.

**Comment by nical** [comment]
Hm actually it isn't that simple because with this API in this situation the end point isn't defined either, so we can either hard-code the start angle to be 0 or skip the arc completely.

**Comment by nical** [comment]
Fixed in lyon_path 0.17.2.
